### PR TITLE
fix: support utf-8 encoded strings in form data

### PR DESCRIPTION
**Pull request by NiclasHaderer**
# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [x] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [x] Have you updated the reference documentation?

Currently UTF-8 form data cannot be parsed and results in an internal server error.  
I added a test which test that and included the fix.

### CODE REVIEW

**Review by Goldziher — COMMENTED** [review]
Hi, thanks. 

The problem with the PR is that you are applying the fix on the query string parser. This is used for regular query strings as well, and your fix will hurt performance. 

Instead, move the fix to `parse_url_encoded_form_data`, this will be better performance wise.

**Comment by NiclasHaderer** [comment]
So should I just create a second function which is only used for body parsing or what do you have in mind?

**Comment by Goldziher** [comment]
> So should I just create a second function which is only used for body parsing or what do you have in mind?

Sure

File: starlite/parsers.py [hunk 1]
```diff
@@ -35,19 +35,37 @@ def parse_query_string(query_string: bytes, encoding: str = "utf-8") -> Tuple[Tu
...
+        encoding: The encoding to use.
+
     Returns:
         A tuple of key value pairs.
     """
     _bools = {"true": True, "false": False, "True": True, "False": False}
     return tuple(
         (k, v if v not in _bools else _bools[v])
-        for k, v in parse_qsl(str(query_string, encoding=encoding), keep_blank_values=True, encoding=encoding)
+        for k, v in parse_qsl(str(form_query_string, encoding=encoding), keep_blank_values=True, encoding=encoding)
```
Comment (Goldziher) [inline starlite/parsers.py:62]: why do you need to pass encoding to `str`, and then again to `parse_qsl`?

Comment (NiclasHaderer) [reply starlite/parsers.py:62]: Good question. That's a mistake on my side.
I'll remove the encoding from `parse_qsl`

**Comment by Goldziher** [comment]
@all-contributors add @NiclasHaderer code

**Pull request merged**